airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -357,7 +360,7 @@ public WebBackendConnectionRead webBackendGetConnection(final WebBackendConnecti
       connection.setStatus(refreshedCatalog.get().getConnectionStatus());
     } else if (catalogUsedToMakeConfiguredCatalog.isPresent()) {
       // reconstructs a full picture of the full schema at the time the catalog was configured.
-      syncCatalog = updateSchemaWithDiscovery(configuredCatalog, catalogUsedToMakeConfiguredCatalog.get());
+      syncCatalog = updateSchemaWithDiscovery(configuredCatalog, catalogUsedToMakeConfiguredCatalog.get(), catalogUsedToMakeConfiguredCatalog.get());
```

Comment:
Why are we just passing the same `catalogUsedToMakeConfiguredCatalog.get()` in twice?

Reply:
Oh hm, I think I see why but it's confusing, maybe in the case where there isn't a refreshed catalog, we should call a different, simpler method that takes advantage of the fact that the original discovered catalog is equal to the discovered catalog?

Reply:
Yeah I can see how this is confusing.

It wouldn't actually simplify much to have two different methods - essentially it's _mostly_ saying "reconstruct it by looking at what we persisted versus the latest catalog". This PR adds a third element: (1) what we persisted; (2) the whole catalog when we persisted it; (3) the latest catalog. It's just that in the case where the catalog hasn't been rediscovered, (2) and (3) are actually the same. But otherwise the logic stays the same.

I do agree that it stands out as _weird_ to pass the same thing twice like that, so I added a helper method in the hopes of better readability.

WDYT?

Reply:
Nice! Makes sense, the helper makes it much clearer